### ISSUE TITLE
Support in Arch

### ISSUE THREAD

**Issue by fridy1307** [regolith-linux/voulage#6]
Not an issue. Just wanted to know if it supports Arch Linux and if so will it automatically install i3 or needed to be installed prior to installing it?

**Comment by kgilmer** [comment]
Hi @fridy1307 , @gardotd426 maintains an Arch version of Regolith.  Hopefully at some point we can get all distro support integrated at this level.

**Comment by gardotd426** [comment]
@fridy1307, Arch-specific questions or issues should be asked at my repo, https://github.com/gardotd426/regolith-de

Regolith DE is on the AUR, so that is NOT where you grab PKGBUILDs from. Ever. Those PKGBUILDs are for internal development and not always in line with the current AUR releases, which are basically "guaranteed to build and work, unless an update breaks something at which point it's reported and I fix it," same as all other AUR packages. 

As for voulage, yes support WILL be coming to Arch. The problem is I caught COVID for the second time, only this time instead of just losing my sense of smell for two weeks, I had a fever of 103+ (Fahrenheit) for 4 days and almost had to be hospitalized. Then I found out that oh, Regolith 2.0 has dropped and everyone is wondering where it's at on the AUR, because they don't realize that I basically have to rewrite the entire 600 lines of code involved, regrab and assign all the sha256sums in order, delete packages that are no longer in existence, add new packages, remove patches that aren't needed anymore, add new patches that *are* needed, etc. So. It'll be a little while, though I am putting in at minimum 8 hour workdays toward it now that I'm healthy.

Part of this is my fault for hating Slack so much and not pushing Ken to maybe have a second option (or just move work discussions altogether) - side note @kgilmer, I'll email you about that later, I have some ideas. 

Anyway, I've been pushing updates to keep v1.6 working after 2.0 changes broke some stuff, and slowly moving through the PKGBUILD and moving stuff to 2.0 on a separate branch, and as soon as it's ready it'll be live. 

2.0 is essentially an entire reworking from the ground up on how Regolith functions in several very important ways. I'm basically having to re-port the DE from scratch, by myself, on a volunteer basis. Only when I wrote the original port, there was no one really waiting for me, because I hadn't announced anything until I was finished. But you can look through the closed issues on the GH repo and see how quickly I fix and update things that are actually issues on my end, and once 2.0 is done, updates will basically be live on Arch within days of being live upstream until another major release like 3.0 comes out. 

>if so will it automatically install i3 or needed to be installed prior to installing it?

i3 is a dependency of Regolith. Obviously it's installed prior to building the PKGBUILD. We use i3-gaps specifically. 

Just to clarify, Regolith DE is already live on the AUR, it has been for over a year I believe. It's just that it's currently on version 1.6, the last release before 2.0. But you can install Regolith DE on Arch right now. You'll just have to wait to get to use all the fun new amazing tools Ken has created for 2.0. That's all. 

Anyway, you have my GH repo for issues with the AUR package, so we will move off this thread

@kgilmer I might add a blurb to the regolith-desktop main README about Arch questions, until we can figure out a more permanent solution. My original plan was to take the opportunity that the move to 2.0 would give me to also move completely away from .debs in every possible instance, and only use git repositories for all packages that I'm grabbing from Regolith (so basically everything that we don't use native Arch repo packages for). I'd already done this for i3xrocks, regolith-i3xrocks-config, remontoire, and a couple others I believe, but I was gonna go all in. But given the scope of all that, I'm just gonna use .debs because your new system and non-launchpad regolith.io repo for .deb downloads including your manifest.txt files have allowed me to automate grabbing all the .debs and their sha256sums, which is a godsend. And since the upstream repos are provided, I can replace those piecemeal once we get 2.0 live on the AUR. 

I believe then we might be able to work on a truly distro-agnostic build system, where you can build the entire desktop environment from source for the distro of your choice using a shell script. This already exists and is integral to an extremely popular set of repos in the gaming enthusiast space, https://GitHub.com/frogging-family. It's operated by the god known as Tk-Glitch, and I've been helping out behind the scenes and contributing patches when I can for several of those repos, and if you look at the `linux-tkg` repo, and check `customization.cfg`, you'll see that you are given a list of distros that you want to choose from, you choose your distro family, and then when you run the `install.sh` script, it will take that choice, pull in any dependencies for that distro using the package manager for that distro, and build everything as it should be. 

This is what I envision for Regolith. A barely interactive script that will pull your distro family from a config file (or a prompt if left empty), then follow a script to install all dependencies with the package manager, then clone all necessary git repos, build what's needed, and create a list of either .pkg.tar, .deb, or .rpm packages comprising the desktop environment. We could even have the user choose "minimal" or "full," etc. 

Seriously check out the frogging-family repo when you get time, especially the linux-tkg and wine-tkg-git repos, you'll see exactly what I mean. Only for us it would probably even be much easier. I'll email you later about suggestions for communication platforms aside from slack, and I'll try to get back into the Slack thing but I'm going to be hitting the grindstone hard moving to 2.0

**Comment by kgilmer** [comment]
I'm sorry to hear that you have been ill and hope you're feeling better now @gardotd426 .  I think the first thing we should address is the collaboration tool issue.  I know other people don't like Slack either.  I use it in my day job and don't really mind either way.  Would GitHub discussions be better?  Something else?

I'll take a look at https://github.com/frogging-family, thanks for the reference.
